pkg/operator/certrotation/target.go

```diff
@@ -135,23 +147,56 @@ func (c RotatedSelfSignedCertKeySecret) EnsureTargetCertKeyPair(ctx context.Cont
 	return targetCertKeyPairSecret, nil
 }
 
-func needNewTargetCertKeyPair(annotations map[string]string, signer *crypto.CA, caBundleCerts []*x509.Certificate, refresh time.Duration, refreshOnlyWhenExpired bool) string {
+func needNewTargetCertKeyPair(currentSecretCert *x509.Certificate, annotations map[string]string, signer *crypto.CA, caBundleCerts []*x509.Certificate, refresh time.Duration, refreshOnlyWhenExpired bool) string {
```

Comment:
nit: I think `currentCert` should do, I'm getting a bit confused by the `secret` part :slightly_smiling_face:

Reply:
fixed